### PR TITLE
Refactor and add volume scheduling tests

### PR DESCRIPTION
**Pull request by msau42**
**What this PR does / why we need it**:
* Combines pv node affinity negative tests into the same file as the rest of volume scheduling tests
* Adds some more test cases around binding failure scenarios

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially address #56100

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by msau42** [comment]
@kubernetes/sig-storage-pr-reviews 
/assign @jsafrane

File: test/integration/scheduler/volume_binding_test.go [hunk 1]
```diff
@@ -135,28 +176,49 @@ func TestVolumeBinding(t *testing.T) {
...
+				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
+			}
+		}
 
 		// Create Pod
 		if _, err := config.client.CoreV1().Pods(config.ns).Create(test.pod); err != nil {
 			t.Fatalf("Failed to create Pod %q: %v", test.pod.Name, err)
 		}
-		if err := waitForPodToSchedule(config.client, test.pod); err != nil {
+		err := waitForPodToSchedule(config.client, test.pod)
```
Comment (bsalamat) [inline test/integration/scheduler/volume_binding_test.go]: In order to make tests run faster, wait for the pod to be marked unschedulable when `test.shouldFail` is true. You can do so by:
```go
err = wait.Poll(pollInterval, wait.ForeverTestTimeout, podUnschedulable(config.client, test.pod.Namespace, test.pod.Name))
```

File: test/integration/scheduler/volume_binding_test.go [hunk 2]
```diff
@@ -222,6 +284,191 @@ func TestVolumeBindingStress(t *testing.T) {
...
+		if !strings.Contains(p.Status.Conditions[0].Message, "node(s) didn't match node selector") || !strings.Contains(p.Status.Conditions[0].Message, "node(s) had volume node affinity conflict") {
+			t.Fatalf("Failed as Pod's %s failure message does not contain expected message: node(s) didn't match node selector, node(s) had volume node affinity conflict", podName)
+		}
+		if err := config.client.CoreV1().Pods(config.ns).Delete(podName, &metav1.DeleteOptions{}); err != nil {
+			t.Fatalf("Failed to delete Pod %s: %v", podName, err)
+		}
+	}
+}
+
+func setupNodes(t *testing.T, nsName string, numberOfNodes int) *testConfig {
```
Comment (bsalamat) [inline test/integration/scheduler/volume_binding_test.go]: isn't `setupCluster` a better name?

File: test/integration/scheduler/volume_binding_test.go [hunk 3]
```diff
@@ -51,78 +67,103 @@ var (
...
-	labelKey   = "kubernetes.io/hostname"
-	labelValue = "node-1"
-	nodeName   = "node1"
-	podLimit   = 100
-	volsPerPod = 5
+	node1            = "node-1"
+	node2            = "node-2"
+	podLimit         = 100
+	volsPerPod       = 5
+	affinityLabelKey = "kubernetes.io/hostname"
```
Comment (bsalamat) [inline test/integration/scheduler/volume_binding_test.go]: maybe nodeAffinityLabelKey?

**Review by bsalamat — COMMENTED** [review]
I had a quick look and it generally looks good. I have several minor comments for now.

**Comment by msau42** [comment]
Comments addressed

**Pull request merged**